server/services/voip/connector/asterisk/ami/PJSIPEndpoint.ts

```diff
@@ -0,0 +1,200 @@
+ *    nformation in different events. event |endpointdetail| indicates
+ *    endpoint name, devicestate etc. |authdetail| indicates type of authentication, password and other
+ *    auth related information. At the end of the series of these events, Asterisk sends |endpointdetailcomplete|
+ * 	  event. At this point of time, promise, on which the consumer is waiting can be resolved.
+ * 6. This class could use asynchronous callbacks. But because the connector will be used extensively by REST layer
+ *    we have taken promise based approach. Promise is returned as a part of executeCommand. Caller would wait for
+ *    the completion/rejection. This class will reject the promise based on error or resolve it in <command>complete
+ *    event.
+ * 7. Important to note that the intermediate events containing a result part for an execution of a particular command
+ *    have same actionid, which is received by this class as a successful execution of a command in actionResultCallback.
+ */
+import { Command, CommandType } from '../Command';
+import { Logger } from '../../../../../../lib/Logger';
+import { Commands } from '../Commands';
+import { EndpointState, IVoipExtensionConfig, IVoipExtensionBase } from '../../../../../../definition/IVoipExtension';
+
+export class PJSIPEndpoint extends Command {
+	constructor(command: string, parametersNeeded: boolean) {
+		super(command, parametersNeeded);
+		this._type = CommandType.AMI;
+		this.logger = new Logger('Endpoint');
+	}
+
+	private getState(endpointState: string): EndpointState {
+		let state: EndpointState = EndpointState.UNKNOWN;
+		if (endpointState.toLowerCase() === 'unavailable') {
+			state = EndpointState.UNREGISTERED;
+		} else if (endpointState.toLowerCase() === 'available') {
+			state = EndpointState.REGISTERED;
+		}
+		return state;
+	}
+
+	/**
+	 * Event handler for endpointlist event containing the information of the endpoints.
+	 * @remark
+	 * This event is generated as a result of the execution of |pjsipshowendpoints|
+	 */
+	onEndpointList(event: any): void {
+		if (event.actionid !== this.actionId) {
+			this.logger?.error('onEndpointList() Unsual behavior. ActionId does not belong to this object',
+			event.actionid,
+			this.actionId);
+			return;
+		}
+		const endPoint = {
+			name: event.objectname,
+			state: this.getState(event.devicestate),
+		};
+		const { result } = this;
+		if (result.endpoints) {
+			result.endpoints.push(endPoint);
+		} else {
+			// create an array of endpoints in the result for
+			// the first time.
+			result.endpoints = [];
+			result.endpoints.push(endPoint);
+		}
+		this.logger?.debug(`onEndpointList Data = ${ JSON.stringify(endPoint) }`);
+	}
+
+	/**
+	 * Event handler for endpointlistcomplete event indicating that all the data
+	 * is received.
+	 */
+	onEndpointListComplete(event: any): void {
+		if (event.actionid !== this.actionId) {
+			this.logger?.error('onEndpointListComplete() Unsual behavior. ActionId does not belong to this object',
```

Comment:
Support for this type of logs will be removed, would you mind changing it to use an object instead?
```js
this.logger.error({ msg: 'onEndpointListComplete() Unsual behavior. ActionId does not belong to this object', event })
```

Reply:
Okay. I was not aware of that there is a logger implementation on the server side. So I reused what I created for the client side. I will rename the client side logger.